src/vcpkg/postbuildlint.cpp

```diff
@@ -891,9 +914,9 @@ namespace vcpkg::PostBuildLint
         const auto release_bin_dir = package_dir / "bin";
 
         std::vector<Path> debug_libs = fs.get_regular_files_recursive(debug_lib_dir, IgnoreErrors{});
-        Util::erase_remove_if(debug_libs, NotExtensionCaseInsensitive{".lib"});
+        Util::erase_remove_if(debug_libs, NotExtensionsCaseInsensitive{{".lib", ".so", ".a", ".dylib"}});
```

Comment:
This will break cross-compilation from Windows to Linux (e.g. Android).

I think the cleanest approach will be to split it per target (not host) instead of attempting some merged path.